### PR TITLE
feat(runtime): beta runtime as client

### PR DESCRIPTION
**Pull request by jonaro00**

### CODE REVIEW

File: runtime/src/beta.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            .await
+        {
+            Ok(o) => o,
+            Err(e) => {
+                eprintln!("Runtime Provisioning phase failed: {e}");
+                exit(131);
+            }
+        };
+        *bytes = serde_json::to_vec(&o).expect("to serialize struct");
+    }
```
Comment (oddgrd) [inline runtime/src/beta.rs:125]: This needs to be polled until all the resources are ready.

File: service/src/runner.rs [hunk 2]
```diff
@@ -12,17 +11,13 @@ use tracing::info;
 pub async fn start(
     beta: bool,
     port: u16,
-    // only used on beta. must match port.
-    address: SocketAddr,
     runtime_executable: PathBuf,
     project_path: &Path,
 ) -> anyhow::Result<(process::Child, runtime::Client)> {
```
Comment (oddgrd) [inline service/src/runner.rs:16]: Will this break anything? For example if someone upgrades deployer, but not runtime? :thinking:

Comment (oddgrd) [reply service/src/runner.rs:16]: Only the args passed to beta runtime changes, though, so it should be fine..

Comment (jonaro00) [reply service/src/runner.rs:16]: Yeah that line was added as part of adding the beta runtime flag

File: runtime/src/beta.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        .expect("invalid shuttle environment");
+    let ip = std::env::var("SHUTTLE_RUNTIME_IP")
+        .expect("runtime ip env var")
+        .parse()
+        .expect("invalid ip");
+    let port = std::env::var("SHUTTLE_RUNTIME_PORT")
+        .expect("runtime port env var")
+        .parse()
+        .expect("invalid port");
+    let api_url = std::env::var("SHUTTLE_API").expect("api env var");
```
Comment (oddgrd) [inline runtime/src/beta.rs:42]: nit: Maybe we could still use a Config struct in a separate file, that is created by getting all these env vars? To clean up the function a bit.

**Review by oddgrd — APPROVED** [review]

**Pull request merged**